server/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -729,6 +718,35 @@ private String retrieveValueToken(ValueIsTokenContext ctx) {
     return addTreeNode(ctx, ProcedureDivisionBodyNode::new);
   }
 
+  @Override
+  public List<Node> visitConstantName(ConstantNameContext ctx) {
+    String subroutineName = PreprocessorStringUtils.trimQuotes(ctx.getText()).toUpperCase();
+    return getLocality(ctx.getStart())
+        .map(
+            locality -> {
+              if (!subroutineService.getUri(subroutineName).isPresent()) {
+                reportSubroutineNotDefined(subroutineName, locality);
+              }
+              subroutineDefinitionMap.putIfAbsent(
+                  subroutineName,
+                  new SubroutineDefinition(
+                      getSubroutineLocation(
+                              new ImmutablePair<>(
+                                  subroutineName, subroutineService.getUri(subroutineName)))
+                          .stream()
+                          .findFirst()
+                          .orElse(
```

Comment:
It should be `orElseGet( () -> new Locality...` to avoid creating a locality when we don't need it.

Reply:
We need to define the locality in order to define SubroutineDefinitionand later collect the Subroutine name usages under that definition.

Reply:
No, I mean, that the code inside `orEsle()` will always be executed even when the positive scenario has passed and we don't need the mock locality. So, it should contain only objects that have already been created. In this particular case `orElseGet()` will work better because it doesn't execute the code when it is not needed.

Reply:
Got it. Thanks :-)